zio-http/jvm/src/test/scala/zio/http/endpoint/openapi/OpenAPIGenSpec.scala

```diff
@@ -2989,6 +2997,114 @@ object OpenAPIGenSpec extends ZIOSpecDefault {
                              |""".stripMargin
         assertTrue(json == toJsonAst(expectedJson))
       },
+      test("Optional ADT payload") {
+        val endpoint     = Endpoint(GET / "static").in[WithOptionalAdtPayload]
+        val generated    = OpenAPIGen.fromEndpoints("Simple Endpoint", "1.0", endpoint)
+        val json         = toJsonAst(generated)
+        val expectedJson =
+          """{
+            |  "openapi" : "3.1.0",
+            |  "info" : {
+            |    "title" : "Simple Endpoint",
+            |    "version" : "1.0"
+            |  },
+            |  "paths" : {
+            |    "/static" : {
+            |      "get" : {
+            |        "requestBody" :
+            |          {
+            |          "content" : {
+            |            "application/json" : {
+            |              "schema" :
+            |                {
+            |                "$ref" : "#/components/schemas/WithOptionalAdtPayload"
+            |              }
+            |            }
+            |          },
+            |          "required" : true
+            |        }
+            |      }
+            |    }
+            |  },
+            |  "components" : {
+            |    "schemas" : {
+            |      "One" :
+            |        {
+            |        "type" :
+            |          "object",
+            |        "properties" : {}
+            |      },
+            |      "SealedTraitCustomDiscriminator" :
+            |        {
+            |        "oneOf" : [
+            |          {
+            |            "$ref" : "#/components/schemas/One"
+            |          },
+            |          {
+            |            "$ref" : "#/components/schemas/Two"
+            |          },
+            |          {
+            |            "$ref" : "#/components/schemas/Three"
+            |          }
+            |        ],
+            |        "discriminator" : {
+            |          "propertyName" : "type",
+            |          "mapping" : {
+            |            "One" : "#/components/schemas/One",
+            |            "Two" : "#/components/schemas/Two",
+            |            "three" : "#/components/schemas/Three"
+            |          }
+            |        }
+            |      },
+            |      "WithOptionalAdtPayload" :
+            |        {
+            |        "type" :
+            |          "object",
+            |        "properties" : {
+            |          "optionalAdtField" : {
+            |            "anyOf": [
+            |              { "type": "null" },
+            |              { "$ref": "#/components/schemas/SealedTraitCustomDiscriminator" }
+            |            ]
+            |          }
```

Comment:
We now correctly have the `anyOf`

Reply:
I wonder, if this should be anyOf, or if the field should just be missing from the required fields list. I'd tend to the second. Most Scala json serialization does not write the field if the option is None

Reply:
Maybe both. To be accept all possible encodings

Reply:
I made the change. Now, we have both 🙂

Reply:
Done. They are now both nullable and non-required